buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json

```diff
@@ -1,5 +1,28 @@
 {
   "build": [
+    {
+      "environment": {},
+      "enabled": true,
+      "continueOnError": true,
+      "alwaysRun": false,
+      "displayName": "Run Begin.sh",
+      "timeoutInMinutes": 0,
+      "condition": "succeeded()",
+      "refName": "TaskRunBegin",
+      "task": {
+        "id": "10f1f9a1-74b0-47ab-87bf-e3c9c68e8b0d",
+        "versionSpec": "0.*",
```

Comment:
Is that the correct version spec? How are you testing these changes? Can you please create a sample definition and verify that these tasks work?

Reply:
@jcagme told me that all I had to do was to paste this on the file :-/ ; Thanks for the suggestion. I am going to create a sample definition.

Reply:
@jcagme helped me to test it: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1552684

Reply:
Thanks. Now please take into account the AgentToolsPath and you should be set. I suggest testing it on a definition where you can put a script there to verify it gets discovered and ran when it is present as well.

Reply:
Yes, we need to add $(AgentToolsPath) to the script so when we deploy the files to the machines this will actually execute what we attempt to execute

Reply:
You can use the branch/build definition used yesterday to drop a test script. Probably you already did this since you asked me this yesterday :)

Reply:
What's the name of this var in Linux/OSX? I tried $AgentToolsPath and $AGENTTOOLSPATH but no success.

Reply:
What do you mean by no success? How does it fail?

Reply:
It seems that these variables don't exist or are empty.

Reply:
@ilyas1974 is there a path already defined for AgentToolsPath in Mac/Linux? As @JohnTortugo points out there is "AGENT_TOOLSDIRECTORY=/Users/buildagent/agent/_work/_tool". Do you recommend using this or we should create a new one?

Reply:
@jcagme I thought the plan was to pass in the value in the definition because we didn't have a great way to set them on the agent machine. Is that not the case?

Reply:
We agreed on adding this environment variable on agent imaging. This is currently what happens on Windows agents and doing the same for Linux/Mac should be straight forward. @ilyas1974 ?

Reply:
@jcagme it could be a system wide environment variable or a property passed in through the root build definition. Either way it should be consistently named AgentToolsDirectory.

Reply:
Yes, it will be. When @ilyas1974 did the work on Windows agents everything was documented so the same work done on different agents (which ever the OS) will be the same.

Reply:
Hey, @JC, is there anything I can do to get this moving again?

Reply:
Sent @ilyas1974 an email since I don't know if he gets notifications about tags here

Reply:
@jcagme talked to @ilyas1974 about adding this new environment variable in Linux / OSX. The request is being tracked here: https://devdiv.visualstudio.com/web/wi.aspx?pcguid=011b8bdf-6d56-4f87-be0d-0092136884d9&id=597942

Reply:
@ilyas1974 said that due his team current demands it may take until the end of the month to get the env. vars on Linux / OSX set. I spoke with @jcagme and we are considering moving these PR forward without the begin/end.sh scripts on Linux / OSX. 

I am going to start testing the changes on Windows. @weshaggard , does that sounds good to you?

Reply:
It is fine if the env variables aren't on the machines yet. You can test these by setting the property when you queue the build.

Reply:
Hi @weshaggard , I've tested this on Windows, Linux and OSX now. I uploaded an executable script from a Linux VM and tested, it worked. I'd advise to still keep the "chmod +x" command before trying to execute the script, though. Just in case someone forget to +x it before uploading, or upload it from a Windows box.

Build where I tested it: 
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/index?buildId=1586143&_a=summary 
[Linux] https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1586178
[Windows] https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1586161
[Windows] https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1586158
[OSX] https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1586157
[Linux] https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1586154

If this sounds good for you I'll apply the patch on the other PRs.

Reply:
Thanks for testing but unless I'm missing something I don't see those updates to this PR to review. 

As far as I can tell from your test runs you didn't even call chmod:
```
if [ -f "$AGENTTOOLSPATH/begin.sh" ]; then ls -la $AGENTTOOLSPATH/begin.sh ; $AGENTTOOLSPATH/begin.sh ; else echo "$AGENTTOOLSPATH/begin.sh script does not exist. Moving on." ; fi
```

At any rate update the PR with what you have tested and I will take a closer review.

Reply:
That was tested in another branch. I didn't include the chmod +x because, as you asked, I uploaded the file (begin.sh) from a Linux machine after locally setting chmod on it. I'll replicate the buildefinition changes on this PR.